internal/services/storagemover/storage_mover_source_endpoint_resource.go

```diff
@@ -0,0 +1,257 @@
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := endpoints.Endpoint{
+				Name: utils.String(model.Name),
+				Properties: endpoints.NfsMountEndpointProperties{
+					Export:     model.Export,
+					Host:       model.Host,
+					NfsVersion: &model.NfsVersion,
+				},
+				Type: utils.String(string(endpoints.EndpointTypeNfsMount)),
+			}
+
+			if model.Description != "" {
+				nfsProperties := properties.Properties.(endpoints.NfsMountEndpointProperties)
+				nfsProperties.Description = utils.String(model.Description)
+				properties.Properties = nfsProperties
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r StorageMoverSourceEndpointResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StorageMover.EndpointsClient
+
+			id, err := endpoints.ParseEndpointID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model StorageMoverSourceEndpointModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			properties := resp.Model
+			if properties == nil {
+				return fmt.Errorf("retrieving %s: model was nil", *id)
+			}
+
+			if metadata.ResourceData.HasChange("description") {
+				nfsProperties := properties.Properties.(endpoints.NfsMountEndpointProperties)
+				nfsProperties.Description = utils.String(model.Description)
+				properties.Properties = nfsProperties
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, *id, *properties); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r StorageMoverSourceEndpointResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.StorageMover.EndpointsClient
+
+			id, err := endpoints.ParseEndpointID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := StorageMoverSourceEndpointModel{
+				Name:           id.EndpointName,
+				StorageMoverId: storagemovers.NewStorageMoverID(id.SubscriptionId, id.ResourceGroupName, id.StorageMoverName).ID(),
+			}
+
+			if model := resp.Model; model != nil {
+				nfsProperties := model.Properties.(endpoints.NfsMountEndpointProperties)
```

Comment:
we're not type checking this, if the cast fails this'll panic

Reply:
Fixed.